### ISSUE TITLE
Importlib-metadata version for python 3.7

### ISSUE THREAD

**Issue by astafan8** [ipython/ipykernel#705]
what is the reason for this ``'importlib-metadata<4;python_version<"3.8.0"',`` requirement for version ``6.0.0``? perhaps it could be relaxed to ``importlib-metadata<=4`` to include ``4``?

**Comment by Carreau** [comment]
no, the api of version 4 is different. This was added to fix #645. Basically this mean any 3.x.y but not any 4.z.t,
with the `<=` we would allow 4.0.0 but not 4.0.1 which is weird.

**Comment by astafan8** [comment]
thank you for the response! will be looking forward to importlib<5.0.0 support for python 3.7 :)
